modules/HomePage/components/TaskForm/TaskForm.tsx

```diff
@@ -25,25 +25,39 @@ const TaskForm: React.FC<TaskFormProps> = ({
   const [priority, setPriority] = useState<number>(2);
   const [dueDate, setDueDate] = useState<Date | null>(null);
 
-  const handleSubmit = (e: React.FormEvent) => {
+  const [addTodo] = useAddTodoMutation();
+
+  const handleSubmit = async (e: React.FormEvent) => {
     e.preventDefault();
     if (!title || !description || !priority || !dueDate) {
       alert("Please fill in all fields");
       return;
     }
+
+    const priorityMapping: { [key: number]: 'HIGH' | 'MEDIUM' | 'LOW' } = {
+      1: 'HIGH',
+      2: 'MEDIUM',
+      3: 'LOW',
+    };
     const newTask: Task = {
-      id: new Date().toISOString(),
       title,
       description,
-      priority,
-      dueDate,
+      priority: priorityMapping[priority],
+      dueDate: dueDate,
       completed: false,
+ 
     };
-    addTask(newTask);
-    setTitle("");
-    setDescription("");
-    setPriority(2);
-    setDueDate(null);
+    try {
+      console.log(newTask);
+      const result = await addTodo(newTask).unwrap();
+      console.log(result);
```

Comment:
Let's remove console logs.
Same comment for similar instances throughout the PR.